extensions/ql-vscode/src/view/results/ResultTablesHeader.tsx

```diff
@@ -0,0 +1,141 @@
+import * as React from "react";
+import { useCallback, useEffect } from "react";
+import { vscode } from "../vscode-api";
+import { openFile, tableHeaderItemClassName } from "./result-table-utils";
+import { sendTelemetry } from "../common/telemetry";
+import {
+  ALERTS_TABLE_NAME,
+  ParsedResultSets,
+} from "../../common/interface-types";
+import { basename } from "../../common/path";
+
+interface Props {
+  queryName: string;
+  queryPath: string;
+  parsedResultSets: ParsedResultSets;
+  selectedTable: string;
+}
+
+export function ResultTablesHeader(props: Props) {
+  const { queryPath, queryName, parsedResultSets, selectedTable } = props;
+
+  const [selectedPage, setSelectedPage] = React.useState(
+    `${parsedResultSets.pageNumber + 1}`,
+  );
+  useEffect(() => {
```

Comment:
Interestingly this effect hook might not strictly be needed. My brain says it should be needed to update `selectedPage`, but when I try removing it everything still works. 🤔 

The case I'd expect to be different is when moving between query results. If I change from viewing the results of one query to a different query, then `parsedResultSets` is updated and we want `selectedPage` to also be reset.

Possibly it's because `ResultTables` is constructing a new copy of this component on every render, so we're always getting the initial calculated value. This could change as we continue to tweak `ResultTables`.

Reply:
I think it's worth keeping it - it does seem like the right thing to do.